src/index.ts

```diff
@@ -2341,7 +2341,12 @@ export class GossipSub extends EventEmitter<GossipsubEvents> implements Initiali
         const backoff = this.backoff.get(topic)
         for (const id of shuffledPeers) {
           const peerStreams = this.peers.get(id)
-          if (peerStreams && hasGossipProtocol(peerStreams.protocol) && !peers.has(id) && !this.direct.has(id)) {
+          if (
+            peerStreams &&
+            this.multicodecs.includes(peerStreams.protocol) &&
```

Comment:
includes loops over an array, would it be better to use a Set?

Reply:
The `multicodecs` field comes from the `PubSub` interface in `@libp2p/interface-pubsub`: https://github.com/libp2p/js-libp2p-interfaces/blob/9f8d5f45647e3bd8943acfbfb77424b0f83c27cf/packages/interface-pubsub/src/index.ts#L74

I see 3 solutions:

1. Modify the pubsub interface and then gossipsub to change `multicodecs` to a `Set<string>`.
2. Add another field on `GossipSub` that is a `Set` and implement `multicodecs` as a getter to conform to the `PubSub` interface,
3. Leave it as it is.

Happy to proceed with either.

In this instance, I do not expect a library consumer to set a high number of value in the array so I am not sure 1 & 2 are worth it.

Reply:
Lets try to upstream the change in the interface (1)

Reply:
> Lets try to upstream the change in the interface (1)

Done: https://github.com/libp2p/js-libp2p-interfaces/pull/260

Waiting for it to be reviewed/merged/released to update this PR.

Reply:
To move forward with this, I need to do a run of the test suite with a `Set` and an `Array` to check if there is any performance gain: https://github.com/libp2p/js-libp2p-interfaces/pull/260#issuecomment-1165474292

Unfortunately, the CI is currently broken on this repo due to the lack of lockfile and discrepancy in the dependencies being used. I am looking if I can fix that myself.

Reply:
Getting there, I hit a road block trying to bump several libp2p libraries, some help would be appreciated :) https://github.com/ChainSafe/js-libp2p-gossipsub/pull/289

Reply:
Running benchmarks on a droplet with `multicodecs` as `Array` and `Set`.

Reply:
Note that the `Set` does not seem to help with performance: https://github.com/libp2p/js-libp2p-interfaces/pull/260#issuecomment-1182823422

Reply:
Would it be acceptable to merge the PR with just 4b873a3bcca1501158728d8a201ceb44f8f55017 ?

Reply:
yeah